File: БР1.1/Захматов Юрий/labs/lab1/src/main/java/com/petproject/itmoacsbackend/property/entities/AmenityEntity.java
package com.petproject.itmoacsbackend.property.entities;


import jakarta.persistence.*;
import lombok.*;

@Table(name = "amenity")
@Entity
@NoArgsConstructor
@Getter
@Setter
@Builder
@AllArgsConstructor
public class AmenityEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false,  unique = true)
    private String name;
}

File: БР1.1/Майстренко Анастасия Николаевна/labs/lab3/rental-microservices/services/catalog/models/amenity.entity.ts
import { Entity, Column, PrimaryGeneratedColumn, ManyToMany, BaseEntity } from 'typeorm';
import { Property } from './property.entity';

@Entity()
export class Amenity extends BaseEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({ type: 'varchar', length: 100, unique: true })
    name: string;

    @Column({ type: 'varchar', length: 100, nullable: true })
    icon: string | null;

    @ManyToMany(() => Property, (p) => p.amenities)
    properties: Property[];
}
